llvm/lib/Transforms/Vectorize/VPlanUnroll.cpp

```diff
@@ -92,18 +92,19 @@ class UnrollState {
   void addRecipeForPart(VPRecipeBase *OrigR, VPRecipeBase *CopyR,
                         unsigned Part) {
     for (const auto &[Idx, VPV] : enumerate(OrigR->definedValues())) {
-      auto Ins = VPV2Parts.insert({VPV, {}});
-      assert(Ins.first->second.size() == Part - 1 && "earlier parts not set");
-      Ins.first->second.push_back(CopyR->getVPValue(Idx));
+      const auto &[V, _] = VPV2Parts.try_emplace(VPV, SmallVector<VPValue *>());
```

Comment:
Hmm, having to explicitly specify `SmallVector<VPValue *>()` seems a bit more verbose than the original code here and below. Is there a way around it?

Reply:
To be honest, I found this inelegant too and wondered if the insert might not be that bad: there seems to be no way around it, unfortunately.

Reply:
Maybe keep the existing code here for now?

Reply:
Can we omit the SmallVector constructor and have try_emplace just use the default constructor?

It would be nice to use try_emplace since it also avoids a copy IIUC.